engine/src/main/scala/cromwell/engine/workflow/tokens/UnhoggableTokenPool.scala

```diff
@@ -0,0 +1,112 @@
+package cromwell.engine.workflow.tokens
+
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicBoolean
+
+import cromwell.core.JobExecutionToken
+import cromwell.core.JobExecutionToken.JobExecutionTokenType
+import cromwell.engine.workflow.tokens.UnhoggableTokenPool.{ComeBackLater, Oink, TokenHoggingLease, UnhoggableTokenPoolResult}
+import io.github.andrebeat.pool._
+
+import scala.collection.immutable.HashSet
+import scala.collection.mutable
+
+final case class UnhoggableTokenPool(tokenType: JobExecutionTokenType) extends SimplePool[JobExecutionToken](
+  capacity = tokenType.maxPoolSize.getOrElse(UnhoggableTokenPool.MaxCapacity),
+  referenceType = ReferenceType.Strong,
+  _factory = () => JobExecutionToken(tokenType, UUID.randomUUID()),
+  _reset = Function.const(()),
+  _dispose = Function.const(()),
+  _healthCheck = Function.const(true)) {
+
+  lazy val hogLimitOption: Option[Int] = tokenType match {
+    case JobExecutionTokenType(_, None, _) => None
+    case JobExecutionTokenType(_, _, 1) => None
+    case JobExecutionTokenType(_, Some(limit), hogFactor) => Some(math.max(1, math.round(limit.floatValue() / hogFactor.floatValue())))
```

Comment:
Will this effectively blow up if someone configured a hogFactor of 0 or a negative one ?